cmd/axelard/cmd/vald/broadcaster/broadcast.go

```diff
@@ -45,11 +45,27 @@ func (b *Broadcaster) Broadcast(ctx sdkClient.Context, msgs ...sdk.Msg) (*sdk.Tx
 
 		response, err = Broadcast(ctx, txf, msgs)
 		if err != nil {
-			// reset account and sequence number in case they were the issue
-			b.txFactory = b.txFactory.
-				WithAccountNumber(0).
-				WithSequence(0)
-			return err
+			if !isABCIError(err) {
+				return err
+			}
+
+			// retry on incorrect account sequence
+			if sdkerrors.ErrWrongSequence.Is(err) {
+				// reset account and sequence number in case they were the issue
+				b.txFactory = b.txFactory.
+					WithAccountNumber(0).
+					WithSequence(0)
```

Comment:
Are they auto-retrieved correctly on the next try?

Reply:
yes, after set to 0

Reply:
yes, after set to 0